src/builder.rs

```diff
@@ -8,11 +8,25 @@ use std::process::Stdio;
 use std::str;
 use std::{fs, io};
 
-use dirs::state_dir;
 use once_cell::sync::OnceCell;
 use tempfile::{Builder, TempDir};
 use tokio::process;
 
+fn state_dir() -> Option<PathBuf> {
+    fn get_absolute_path(path: OsString) -> Option<PathBuf> {
+        let path = PathBuf::from(path);
+        path.is_absolute().then_some(path)
+    }
+
+    if let Some(xdg) = std::env::var_os("XDG_STATE_HOME") {
```

Comment:
note that in dirs, state_dir() always returns None for macos.

I'm not sure if we should follow their impl here, however since macos is also unix, I tend to think it's ok to use ~/.local/state